### PR TITLE
Add valid_move? method to Queen

### PR DESCRIPTION
**Pull request by jonmast**
This returns true if the move is in a straight line and not obstructed.

### CODE REVIEW

**Comment by radiofreemattd** [comment]
Looks good to me...we now are starting to develop the consistency I was looking for and I think the code looks the way you and @mscoutermarsh wanted! :+1:

File: spec/models/queen_spec.rb [hunk 1]
```diff
@@ -4,4 +4,30 @@
   it 'has type of Queen' do
     expect(subject.type).to eq 'Queen'
   end
+
+  describe '#valid_move?' do
+    let(:queen) { FactoryGirl.create(:queen, position_x: 3, position_y: 3) }
```
Comment (radiofreemattd) [inline spec/models/queen_spec.rb]: Don't think you actually need the FactoryGirl preface here.

File: spec/models/queen_spec.rb [hunk 2]
```diff
@@ -4,4 +4,30 @@
...
+    it 'returns true for diagonal moves' do
+      expect(queen.valid_move?(1, 5)).to eq true
+    end
+
+    it 'returns false for moves that are not straight lines' do
+      expect(queen.valid_move?(4, 5)).to eq false
+    end
+
+    it 'returns false if obstructed' do
+      FactoryGirl.create(:pawn, position_x: 3, position_y: 4, game: queen.game)
```
Comment (radiofreemattd) [inline spec/models/queen_spec.rb]: Again, don't need and also don't need to override the game param during create. I don't really care which we do things, but we should standardize one way or the other :smile:

**Comment by mscoutermarsh** [comment]
Looks great. Going to merge. :smile:

The `FactoryGirl` is optional. Usually I decide on going one way or the other. Doesn't matter. Up to you all. Can change it later :)

**Pull request merged**